### PR TITLE
Rust sdk upgrade

### PR DESCRIPTION
**Pull request by oltraj**
Upgraded the Rust SDK

### CODE REVIEW

File: couchbase-sys/Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ cc = "1.0"
 cmake = "0.1.45"
 
 [build-dependencies.bindgen]
-version = "0.55"
+version = "0.62"
```
Comment (chvck) [inline couchbase-sys/Cargo.toml:24]: This should probably be an update to the latest version.

File: couchbase-sys/Makefile [hunk 2]
```diff
@@ -0,0 +1,31 @@
+binding:
```
Comment (chvck) [inline couchbase-sys/Makefile:1]: For my understanding why do we need this makefile? `cargo build` should run all the bindgen bits automatically.

File: couchbase-sys/headers-3.3.10.h [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#include "libcouchbase-3.3.10/include/libcouchbase/couchbase.h"
```
Comment (chvck) [inline couchbase-sys/headers-3.3.10.h:1]: Again for my understanding what does this file do?

File: couchbase-sys/src/lib.rs [hunk 4]
```diff
@@ -8,3 +8,9 @@
 extern crate openssl_sys;
 
 include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
```
Comment (chvck) [inline couchbase-sys/src/lib.rs:11]: Was this meant to be here?

**Review by chvck — CHANGES_REQUESTED** [review]
I think that the commits for this PR should be tidied up. One for updating bindgen version and then another for updating libcouchbase.

File: couchbase-sys/Cargo.toml (hunk 1 above)
Comment (oltraj) [reply couchbase-sys/Cargo.toml:24]: I agree

**Comment by oltraj** [comment]
> I think that the commits for this PR should be tidied up. One for updating bindgen version and then another for updating libcouchbase.

Thank you for your feedback, @chvck. I appreciate your input. I agree that it would be a good idea to tidy up the commits for this PR. I'll create one commit for updating bindgen to the latest version and another for updating libcouchbase. I'll work on this and update the PR accordingly."

File: couchbase-sys/src/lib.rs (hunk 4 above)
Comment (oltraj) [reply couchbase-sys/src/lib.rs:11]: Indeed, there appears to be a duplication of the extern crate openssl_sys; and include!(concat!(env!("OUT_DIR"), "/bindings.rs")); lines in the code. This duplication seems unintentional and can be safely removed. The primary purpose of this code is to set up attributes, dependencies, and include generated Rust bindings. Thank you for pointing this out, and we will make sure to clean it up in the code.

Comment (chvck) [reply couchbase-sys/src/lib.rs:11]: Sorry, I was referring to the commented out unit test below.

**Comment by oltraj** [comment]
> I think that the commits for this PR should be tidied up. One for updating bindgen version and then another for updating libcouchbase.

Created the PR for the two : 
* https://github.com/couchbaselabs/couchbase-rs/pull/150
* https://github.com/couchbaselabs/couchbase-rs/pull/151

**Comment by oltraj** [comment]
Hello @chvck  , please review the created two PR, thank you

**Comment by chvck** [comment]
Closing as this work has now been completed, and merged, in another PR.

**Pull request closed without merging**